### PR TITLE
MGMT-2292 cache payload in UserAuth flow

### PR DESCRIPTION
**Pull request by danielerez**
In order to avoid unnecessary requests to CapabilityReview API in AMS,
store the user payload in cache.

### CODE REVIEW

**Comment by danielerez** [comment]
@filanov @ronniel1 @rollandf

File: pkg/auth/auth_handler.go [hunk 1]
```diff
@@ -179,26 +179,34 @@ func (a *AuthHandler) AuthUserAuth(token string) (interface{}, error) {
...
-	if payload.Username == "" {
-		a.log.Error("Missing username in token")
-		return nil, fmt.Errorf("Missing username in token")
-	}
-
 	return payload, nil
 }
 
 func (a *AuthHandler) storeAdminInPayload(payload *ocm.AuthPayload) error {
+	payloadKey := payload.Username + "_is_admin"
```
Comment (filanov) [inline pkg/auth/auth_handler.go:197]: Is `_is_admin` used somewhere?

Comment (danielerez) [reply pkg/auth/auth_handler.go:197]: Just here, it's only a postfix to make the key unique.

Comment (filanov) [reply pkg/auth/auth_handler.go:197]: Then when you read it don't you need to get this suffix as well?

**Review by filanov — APPROVED** [review]

**Comment by filanov** [comment]
@danielerez this is the test that failed:
```
[2020-09-30T11:50:26.227Z] [91m[1m• Failure [0.104 seconds][0m
[2020-09-30T11:50:26.227Z] test authorization
[2020-09-30T11:50:26.227Z] [90m/home/workspace/assisted-service_PR-445/subsystem/authz_test.go:15[0m
[2020-09-30T11:50:26.227Z]   admin user
[2020-09-30T11:50:26.227Z]   [90m/home/workspace/assisted-service_PR-445/subsystem/authz_test.go:85[0m
[2020-09-30T11:50:26.227Z]     [91m[1mcan get all clusters [It][0m
[2020-09-30T11:50:26.227Z]     [90m/home/workspace/assisted-service_PR-445/subsystem/authz_test.go:86[0m
[2020-09-30T11:50:26.227Z] 
[2020-09-30T11:50:26.227Z]     [91mUnexpected error:
[2020-09-30T11:50:26.227Z]         <*installer.GetClusterNotFound | 0xc0006a8548>: {
[2020-09-30T11:50:26.227Z]             Payload: {
[2020-09-30T11:50:26.227Z]                 Code: "404",
[2020-09-30T11:50:26.227Z]                 Href: "",
[2020-09-30T11:50:26.227Z]                 ID: 404,
[2020-09-30T11:50:26.227Z]                 Kind: "Error",
[2020-09-30T11:50:26.227Z]                 Reason: "record not found",
[2020-09-30T11:50:26.227Z]             },
[2020-09-30T11:50:26.227Z]         }
[2020-09-30T11:50:26.227Z]         [GET /clusters/{cluster_id}][404] getClusterNotFound  &{Code:0xc00061cf50 Href:0xc00061cf60 ID:0xc0005a0784 Kind:0xc00061cf70 Reason:0xc00061cf80}
[2020-09-30T11:50:26.227Z]     occurred[0m
[2020-09-30T11:50:26.227Z] 
[2020-09-30T11:50:26.227Z]     /home/workspace/assisted-service_PR-445/subsystem/authz_test.go:88
```
Can it be related to the fix?

**Comment by danielerez** [comment]
yes, fixed. CI passes now.

File: pkg/auth/auth_handler.go [hunk 2]
```diff
@@ -179,26 +179,35 @@ func (a *AuthHandler) AuthUserAuth(token string) (interface{}, error) {
 		return nil, err
 	}
 
+	if payload.Username == "" {
+		a.log.Error("Missing username in token")
+		return nil, fmt.Errorf("Missing username in token")
```
Comment (filanov) [inline pkg/auth/auth_handler.go:184]: please use errors package instead of fmt, errors have a stuck trace that we can use later.

Comment (danielerez) [reply pkg/auth/auth_handler.go:184]: It's actually not related to this PR, I only moved the existing lines... But sure, will change.

**Review by filanov — APPROVED** [review]

**Pull request merged**